### DISCUSSION TITLE
Typography 可否支持自定义的渲染 Input 组件

### DISCUSSION THREAD

**Discussion by kaysonwu** [ant-design/ant-design#38581] [Ideas]
<Typography.Text editable>2022-11-11</Typography.Text> 点击编辑时可以渲染成 日期组件

**Comment by gregor-mueller** [comment]
I am also looking for this.

Would be great to have something like this:

```tsx
<Typography.Text editable input={ type: 'number', min: 0, max: 10 }>0</Typography.Text>
```

```tsx
<Typography.Text editable input={ type: 'date', min: 0, max: 10 }>2022-11-11</Typography.Text>
```

The `input` prop would accept some exposed props of the target component - depending on the `type` property.

E.g (non-exhaustive):

- `type: "number"`: `min`, `max`, ...
- `type: "text-area"`: `maxLength`, `showCount`, `autoSize`, ... (I would even remove the `TextArea` specific props, like `maxLength` from `Typography.Text`)
- `type: "time-picker"`: `format`, `minuteStep`, ...
- `type: "date-picker"`: `format`, `showTime`, ...
- `type: "date-range-picker"`: `showTime`, `disabled`, ...

**Reply by chen-yan** [reply]
style customize support as well.
